sources/packages/web/src/components/aest/students/DisabilityStatusUpdateTileValue.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <title-value propertyTitle="Disability status">
+    <template #value
+      >{{ disabilityStatusToDisplay(disabilityStatus) }}
+      <v-btn
+        v-if="allowDisabilityStatusUpdate"
+        variant="text"
+        color="primary"
+        @click="showDisabilityStatusModal"
+        ><span class="text-decoration-underline font-bold"
+          >Update disability status</span
+        ></v-btn
+      ></template
+    >
+  </title-value>
+  <update-disability-status-modal
+    ref="updateDisabilityStatusModal"
+    :allowedRole="Role.StudentUpdateDisabilityStatus"
+  />
+</template>
+
+<script lang="ts">
+import { ref, defineComponent, PropType } from "vue";
+import { StudentService } from "@/services/StudentService";
+import { useFormatters, ModalDialog, useSnackBar } from "@/composables";
+import { DisabilityStatus, Role } from "@/types";
+import UpdateDisabilityStatusModal from "@/components/aest/students/modals/UpdateDisabilityStatusModal.vue";
+import { UpdateDisabilityStatusAPIInDTO } from "@/services/http/dto";
+
+export default defineComponent({
+  components: { UpdateDisabilityStatusModal },
+  emits: {
+    disabilityStatusUpdated: (data: UpdateDisabilityStatusAPIInDTO) => {
+      return !!data;
+    },
+  },
+  props: {
+    studentId: {
+      type: Number,
+      required: true,
+    },
+    disabilityStatus: {
+      type: String as PropType<DisabilityStatus>,
+      required: true,
+    },
+    allowDisabilityStatusUpdate: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+  },
+  setup(props, context) {
+    const { disabilityStatusToDisplay } = useFormatters();
+    const updateDisabilityStatusModal = ref(
+      {} as ModalDialog<UpdateDisabilityStatusAPIInDTO | false>,
+    );
+    const snackBar = useSnackBar();
+
+    const showDisabilityStatusModal = async () => {
+      const updatedDisabilityStatus =
+        await updateDisabilityStatusModal.value.showModal();
+      if (updatedDisabilityStatus) {
+        try {
+          await StudentService.shared.updateDisabilityStatus(
+            props.studentId,
+            updatedDisabilityStatus,
+          );
+          snackBar.success("Disability status updated successfully.");
+          context.emit("disabilityStatusUpdated", updatedDisabilityStatus);
```

Comment:
Is the `updatedDisabilityStatus` needed here since the event will call the `StudentService.shared.getStudentProfile` in the consumer?
I am not trying to justify the event args based on the consumer but if there is no need right now I guess that we do not need to provide an event parameter.

Reply:
Nice catch. Forgot to remove that.